File: GestionCours/src/Entity/Cours.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
/**
 * Cours
 *
 * @ORM\Table(name="cours", indexes={@ORM\Index(name="FK_UserNom", columns={"utilisateurNom"})})
 * @ORM\Entity(repositoryClass="App\Repository\CoursRepository")
 *
 */
class Cours
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     * @ORM\Column(name="utilisateurNom", type="string", length=100, nullable=false)
     */
    private $utilisateurnom;

    /**
     * @var string
     * @ORM\Column(name="nomCours", type="string", length=50, nullable=false)
     * @Assert\NotBlank(message="Nom cours is required")

     */
    private $nomcours;

    /**
     * @var int
     * @Assert\NotBlank(message="Nom Cours is required")
     * @ORM\Column(name="nbrChapitres", type="integer", nullable=false)
     *
     */
    private $nbrchapitres;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="text", length=65535, nullable=false)
     */
    private $description;

    /**
     * @var string
     *
     * @ORM\Column(name="coursImg", type="string", length=5000, nullable=false)
     */
    private $coursimg;


    /**
     * @var \Categorie
     *
     * @ORM\ManyToOne(targetEntity=Categorie::class, inversedBy="cours")
     * @ORM\JoinColumn(nullable=false)({
     *   @ORM\JoinColumn(name="categorie_id", referencedColumnName="id")
     * })
     */
    private $Categorie;

    /**
     * @ORM\OneToMany(targetEntity=Chapitre::class, mappedBy="cours", orphanRemoval=true)
     */
    private $Chapitre;

    public function __construct()
    {
        $this->Chapitre = new ArrayCollection();
    }


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUtilisateurnom(): ?string
    {
        return $this->utilisateurnom;
    }

    public function setUtilisateurnom(string $utilisateurnom): self
    {
        $this->utilisateurnom = $utilisateurnom;

        return $this;
    }

    public function getNomcours(): ?string
    {
        return $this->nomcours;
    }

    public function setNomcours(string $nomcours): self
    {
        $this->nomcours = $nomcours;

        return $this;
    }

    public function getNbrchapitres(): ?int
    {
        return $this->nbrchapitres;
    }

    public function setNbrchapitres(int $nbrchapitres): self
    {
        $this->nbrchapitres = $nbrchapitres;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getCoursimg(): ?string
    {
        return $this->coursimg;
    }

    public function setCoursimg(string $coursimg): self
    {
        $this->coursimg = $coursimg;

        return $this;
    }

    public function getCategorienom(): ?string
    {
        return $this->categorienom;
    }

    public function setCategorienom(string $categorienom): self
    {
        $this->categorienom = $categorienom;

        return $this;
    }



    //////////////////////////
    public function getWebpath(){
        return null === $this->coursimg ? null : $this->getUploadDir.'/'.$this->coursimg;
    }
    protected  function  getUploadRootDir(){

        return __DIR__.'/../../../front/web/images'.$this->getUploadDir();
    }
    protected function getUploadDir(){

        return'';
    }

    public function getCategorie(): ?Categorie
    {
        return $this->Categorie;
    }

    public function setCategorie(?Categorie $Categorie): self
    {
        $this->Categorie = $Categorie;

        return $this;
    }


    public function __toString()
    {
        return $this->getNomcours();
    }

    /**
     * @return Collection|Chapitre[]
     */
    public function getChapitre(): Collection
    {
        return $this->Chapitre;
    }

    public function addChapitre(Chapitre $chapitre): self
    {
        if (!$this->Chapitre->contains($chapitre)) {
            $this->Chapitre[] = $chapitre;
            $chapitre->setCours($this);
        }

        return $this;
    }

    public function removeChapitre(Chapitre $chapitre): self
    {
        if ($this->Chapitre->removeElement($chapitre)) {
            // set the owning side to null (unless already changed)
            if ($chapitre->getCours() === $this) {
                $chapitre->setCours(null);
            }
        }

        return $this;
    }













}

File: 9ariniPi/src/entities/Cours.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author fares
 */
public class Cours {

    private int id;
    private String utilisateurNom;
    private String nomCours;
    private int nbrChapitres;
    private String description;
         private String coursImg;
    private int categorie_id;
    List<Chapitres> ch;
    List<Ressources> res;   
    public Cours() {
    }
    public Cours(String utilisateurNom, String nomCours, int nbrChapitres,String description,int categorie_id) {
        this.utilisateurNom = utilisateurNom;
        this.nomCours = nomCours;
        this.nbrChapitres = nbrChapitres;
                this.description=description;

        this.categorie_id=categorie_id;
            ch  = new ArrayList<Chapitres>();
    res  = new ArrayList<Ressources>();

    }
    
    
    public Cours(String utilisateurNom, String nomCours, int nbrChapitres, String description, String coursImg, int categorie_id) {
        this.utilisateurNom = utilisateurNom;
        this.nomCours = nomCours;
        this.nbrChapitres = nbrChapitres;
        this.description = description;
        this.coursImg = coursImg;
        this.categorie_id = categorie_id;
    }

    public Cours(int id, String utilisateurNom, String nomCours, int nbrChapitres, String description, String coursImg, int categorie_id) {
        this.id = id;
        this.utilisateurNom = utilisateurNom;
        this.nomCours = nomCours;
        this.nbrChapitres = nbrChapitres;
        this.description = description;
        this.coursImg = coursImg;
        this.categorie_id = categorie_id;
    }

    public String getCoursImg() {
        return coursImg;
    }

    public void setCoursImg(String coursImg) {
        this.coursImg = coursImg;
    }

    public Cours(String nomCours) {
        this.nomCours = nomCours;
    }
  public Cours(int id) {
        this.id = id;
    }
    public Cours(int id, String utilisateurNom, String nomCours, int nbrChapitres,String description,int categorie_id) {
        this.id = id;
        this.utilisateurNom = utilisateurNom;
        this.nomCours = nomCours;
        this.nbrChapitres = nbrChapitres;
                this.description=description;
        this.categorie_id=categorie_id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getCategorie_id() {
        return categorie_id;
    }

    public void setCategorie_id(int categorie_id) {
        this.categorie_id = categorie_id;
    }

    public String getUtilisateurNom() {
        return utilisateurNom;
    }

    public void setUtilisateurNom(String utilisateurNom) {
        this.utilisateurNom = utilisateurNom;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNomCours() {
        return nomCours;
    }

    public void setNomCours(String nomCours) {
        this.nomCours = nomCours;
    }

    public int getNbrChapitres() {
        return nbrChapitres;
    }

    public void setNbrChapitres(int nbrChapitres) {
        this.nbrChapitres = nbrChapitres;
    }

    @Override
    public String toString() {
        return "Cours{" + "id=" + id + ", utilisateurNom=" + utilisateurNom + ", nomCours=" + nomCours + ", nbrChapitres=" + nbrChapitres + ", description=" + description + ", categorie_id=" + categorie_id + '}';
    }

    public Cours(String utilisateurNom, String nomCours, int nbrChapitres, String description) {
        this.utilisateurNom = utilisateurNom;
        this.nomCours = nomCours;
        this.nbrChapitres = nbrChapitres;
        this.description = description;
    }

 



    

  

}
